### PR TITLE
Endyear

### PR DESCRIPTION
**Pull request by DonMischo**
code changes for endyear
updated tex files

### CODE REVIEW

**Comment by ArchibaldBienetre** [comment]
So, if this were a real job, and you were my intern, your grades would actually be pretty bad. :disappointed: 

I'm a bit disappointed, but I'll spare you the rant. 

I'd ask you kindly to give it a try and fix those failing tests. 

Any git branching / rebasing issues, we'll fix together then after.

**Comment by DonMischo** [comment]
Fixed half of broken tests.

However, I'm not able to fix the heuristic tests... I think I traced down the issue to their roots. 
CsvValidator.java:66 record.size(); counts the empty column, too. So header and contents are always equal in length. Do I miss something?

**Comment by ArchibaldBienetre** [comment]
Ah, sorry, was out sick for a week and missed your (timely) reply. 

I'll have a look asap (likely, Wednesday night as usual), will be on leave next week so I may jump ahead with the rebase.

**Comment by ArchibaldBienetre** [comment]
Feels like work :sweat_smile: 
image

**Comment by ArchibaldBienetre** [comment]
Merge conflicts are a lot, because an already-merged branch has been continued here, it seems: #97

**Comment by ArchibaldBienetre** [comment]
Worst merge I've had in months. I need to reconsider my approach... 

Rebase does not look good, merge is a mess... 

In general, it's always easier if it's one's own code.

:mag: I'll go through the code, probably refactor it a bit (e.g. split [very long lines](https://github.com/SergelsOrg/csv2tex/blob/6c7f39a7254ae3e6b21de4e9ae3fd0882f6ecaa0/src/main/java/org/example/csv2tex/placeholders/schoolspecific/ErfurtSchoolTablePlaceholderReplacer.java#L81)), then re-try the rebase.

File: sample templates/end_year/SchoolReportTemplate_SchoolYear__ZeugnisTemplate_Schuljahr.tex [hunk 1]
```diff
@@ -0,0 +1,84 @@
+\input{SchoolReportTemplate_ZeugnisTemplate_Packages.tex}
+
+\pagenumbering{gobble}
+\setlength\parindent{0pt}
+\setlength\cellspacetoplimit{2pt}
+\setlength\cellspacebottomlimit{2pt}
+
+\newcommand{\tableGradeWidth}{.050\linewidth}
+\newcommand{\tableGradeWidthLevel}{.060\linewidth}
+\newcommand{\tableLineBreak}{\vspace{.2em}\newline}
```
Comment (ArchibaldBienetre) [inline sample templates/end_year/SchoolReportTemplate_SchoolYear__ZeugnisTemplate_Schuljahr.tex:10]: commands should be part of the `...Commands.tex`

Comment (ArchibaldBienetre) [reply sample templates/end_year/SchoolReportTemplate_SchoolYear__ZeugnisTemplate_Schuljahr.tex:10]: ... like it was changed on the main branch in March f08d451d5628601933f2c31a14f18cd01c8e6fc8

**Comment by ArchibaldBienetre** [comment]
Successfully rebased it. 

Now I'll look into the remaining failing tests :mag:

**Comment by ArchibaldBienetre** [comment]
2 hours and 15 minutes of work. But now the tests should run again. 

Now, I can have a closer look at the actual code changes soon... but maybe not tonight anymore.

File: sample templates/end_year/SchoolReportTemplate_SchoolYear__ZeugnisTemplate_Schuljahr.tex (hunk 1 above)
Comment (ArchibaldBienetre) [reply sample templates/end_year/SchoolReportTemplate_SchoolYear__ZeugnisTemplate_Schuljahr.tex]: fixed

File: src/main/java/org/example/csv2tex/csv/CsvValidator.java [hunk 2]
```diff
@@ -99,7 +99,7 @@ private void checkFaultsWithHeuristics(float headerRowLength, Map<Integer, Integ
             }
         }
 
-        // more than 20% / at least 10 rows off --> assume header error
+        // more than 20% / at least 11 rows off --> assume header error
```
[the remark this answers is not in the document]
Comment (ArchibaldBienetre) [reply src/main/java/org/example/csv2tex/csv/CsvValidator.java]: fixed

File: src/main/java/org/example/csv2tex/placeholders/schoolspecific/ErfurtSchoolTablePlaceholderReplacer.java [hunk 3]
```diff
@@ -141,7 +147,7 @@ String makeCompetencyEntriesMS(List<SchoolCompetencyData> competencyList) {
                 appendUnderlined(competency, schoolCompetencyData.schoolSubCompetency);
             }
             if (!schoolCompetencyData.description.isEmpty()) {
-                String temp = schoolCompetencyData.description.replaceAll("\r\n|\r|\n", "\\" + TEX_TABLE_LINE_BREAK);
+                String temp = schoolCompetencyData.description.replaceAll("\r\n|\r|\n", "\\\\" + TEX_TABLE_LINE_BREAK);
```
Comment (ArchibaldBienetre) [inline src/main/java/org/example/csv2tex/placeholders/schoolspecific/ErfurtSchoolTablePlaceholderReplacer.java:150]: may make sense to pull those slashes into the constant instead

File: src/main/java/org/example/csv2tex/placeholders/schoolspecific/ErfurtSchoolTablePlaceholderReplacer.java [hunk 4]
```diff
@@ -119,7 +125,7 @@ String makeTableEntry(List<SchoolCompetencyData> competencyList, String partOfYe
     }
 
     private boolean shouldRenderAsMajorSubject(SchoolCompetencyData firstSchoolCompetencyData, String partOfYear) {
-        return partOfYear.equals("Endjahr") ||
+        return partOfYear.equals("Schuljahr") ||
```
[the remark this answers is not in the document]
Comment (ArchibaldBienetre) [reply src/main/java/org/example/csv2tex/placeholders/schoolspecific/ErfurtSchoolTablePlaceholderReplacer.java:129]: done

**Comment by ArchibaldBienetre** [comment]
3 hours. I'm wiped out :sleepy: 

@DonMischo please double-check the QA side (i.e., please run it and check if it still looks fine).
I don't know if I broke the layout by accident :see_no_evil:

File: sample templates/end_year/SchoolReportTemplate_SchoolYear__ZeugnisTemplate_Schuljahr.tex [hunk 5]
```diff
@@ -31,36 +72,12 @@
...
-	\end{minipage}
-	}%
-	\newpagedefs
-
-	#tables
-
-	\thispagestyle{mypagestyle}
+%	\newpage
+%	\thispagestyle{plain}
+%	\phantom{~}
```
Comment (ArchibaldBienetre) [inline sample templates/end_year/SchoolReportTemplate_SchoolYear__ZeugnisTemplate_Schuljahr.tex:82]: Do we need this? (Clean Code and stuff ;) )

File: sample templates/end_year/SchoolReportTemplate_ZeugnisTemplate_Commands.tex [hunk 6]
```diff
@@ -17,34 +15,41 @@
...
 	\end{tabularx}\vspace{2pt}
 }
 
+% argument is the page threshold until a newline should be added
+\newcommand\newOptionalNewPage[1]{
+	\ifnum #1>\value{page}
+		\newpage
+		\phantom{~}
+	\fi
+}
```
Comment (ArchibaldBienetre) [inline sample templates/end_year/SchoolReportTemplate_ZeugnisTemplate_Commands.tex:52]: Nice! Micha spent quite some time to figure this out - with this, a report can always be 12 pages (with filler pages in between).

**Review by ArchibaldBienetre — APPROVED** [review]

**Review by ArchibaldBienetre — APPROVED** [review]

**Pull request merged**